src/Renci.SshNet/Sftp/SftpFileStream.cs

```diff
@@ -838,29 +818,31 @@ public override long Seek(long offset, SeekOrigin origin)
                     // Abandon the read buffer.
                     _bufferPosition = 0;
                     _bufferLen = 0;
+                }
 
-                    // Seek to the new position.
-                    switch (origin)
-                    {
-                        case SeekOrigin.Begin:
-                            newPosn = offset;
-                            break;
-                        case SeekOrigin.Current:
-                            newPosn = _position + offset;
-                            break;
-                        case SeekOrigin.End:
-                            var attributes = _session.RequestFStat(_handle, false);
-                            newPosn = attributes.Size - offset;
-                            break;
-                    }
-
-                    if (newPosn < 0)
-                    {
-                        throw new EndOfStreamException();
-                    }
+                // Seek to the new position.
+                switch (origin)
+                {
+                    case SeekOrigin.Begin:
+                        newPosn = offset;
+                        break;
+                    case SeekOrigin.Current:
+                        newPosn = _position + offset;
+                        break;
+                    case SeekOrigin.End:
+                        var attributes = _session.RequestFStat(_handle, false);
+                        newPosn = attributes.Size + offset;
+                        break;
+                    default:
+                        throw new ArgumentException("Invalid seek origin.", "origin");
```

Comment:
nit: I like to use named parameters when there is more than one argument with the same type, as the compiler won't catch out of order parameters and human reviewers may not know or remember the order of the arguments.

Reply:
Resolved